src/Layout/toolset-tasks/OverrideAndCreateBundledNETCoreAppPackageVersion.cs

```diff
@@ -56,9 +57,24 @@ public override bool Execute()
 
             propertyGroup.Element(ns + "NETCoreSdkVersion").Value = newSDKVersion;
 
-            var originalBundledNETCoreAppPackageVersion =
-                propertyGroup.Element(ns + "BundledNETCoreAppPackageVersion").Value;
-            propertyGroup.Element(ns + "BundledNETCoreAppPackageVersion").Value = microsoftNETCoreAppRefPackageVersion;
+
+            var originalBundledNETCoreAppPackageVersion = propertyGroup.Element(ns + "BundledNETCoreAppPackageVersion").Value;
+            var parsedOriginalBundledPackageVersion = SemanticVersion.Parse(originalBundledNETCoreAppPackageVersion);
+            var parsedMicrosoftNETCoreAppRefPackageVersion =
+                SemanticVersion.Parse(microsoftNETCoreAppRefPackageVersion);
+
+            // In the case where we have a new major version, it'll come in first through the dotnet/runtime flow of the
+            // SDK's own package references. The Stage0 SDK's bundled version props file will still be on the older major version
+            // (and the older TFM that goes along with that). If we just replaced the bundled version with the new major version,
+            // apps that target the 'older' TFM would fail to build. So we need to keep the bundled version from the existing
+            // bundledversions.props in this one specific case.
+
+            var newBundledPackageVersion =
+                parsedOriginalBundledPackageVersion.Major == parsedMicrosoftNETCoreAppRefPackageVersion.Major
+                ? microsoftNETCoreAppRefPackageVersion
+                : originalBundledNETCoreAppPackageVersion;
+
+            propertyGroup.Element(ns + "BundledNETCoreAppPackageVersion").Value = newBundledPackageVersion;
 
```

Comment:
@dsplaisted would you mind double-checking this and seeing if it matched what you were talking about?

Reply:
Yes, this matches what I expected.  It might be better to use `NuGetVersion` instead of `SemanticVersion`.  I'm not too sure of the differences between them, but the code I've seen has used `NuGetVersion`.